src/frontend-samples/screen-space-effects-sample/ScreenSpaceEffectsWidget.tsx

```diff
@@ -0,0 +1,173 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "common/samples-common.scss";
+import React, { useEffect } from "react";
+import { Slider, Toggle } from "@bentley/ui-core";
+import { cloneEffectsConfig, EffectsConfig, getCurrentEffectsConfig, updateEffectsConfig } from "./Effects";
+import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, UiItemsProvider, WidgetState } from "@bentley/ui-abstract";
+import { Angle } from "@bentley/geometry-core";
+import ScreenSpaceEffectsApi from "./ScreenSpaceEffectsApi";
+import { useActiveViewport } from "@bentley/ui-framework";
+
+export const ScreenSpaceEffectsWidget: React.FunctionComponent = () => {
+  const viewport = useActiveViewport();
+  const [lensDistortionState, setLensDistortionState] = React.useState<boolean>(true);
+  const [saturationState, setSaturationState] = React.useState<boolean>(true);
+  const [vignetteState, setVignetteState] = React.useState<boolean>(true);
+  const [effectsConfigState, setEffectsConfigState] = React.useState<EffectsConfig>(getCurrentEffectsConfig());
+  const [lensAngleState, setLensAngleState] = React.useState<number>(90);
+
+  useEffect(() => {
+    if (!ScreenSpaceEffectsApi._effectsRegistered) {
+      ScreenSpaceEffectsApi.registerEffects();
+      ScreenSpaceEffectsApi._effectsRegistered = true;
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!viewport)
+      return;
+
+    // The lens distortion effect requires the camera to be enabled. Turn it on if it's not already.
+    let lensAngle = lensAngleState;
+    if (!viewport.view.isCameraEnabled())
+      viewport.turnCameraOn(Angle.createDegrees(lensAngle));
+    else
+      lensAngle = viewport.view.camera.getLensAngle().degrees;
+
+    setLensAngleState(lensAngle);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [viewport]);
+
+  // Handle Lens angle change
+  useEffect(() => {
+    if (!viewport)
+      return;
+
+    viewport.turnCameraOn(Angle.createDegrees(lensAngleState));
+
+  }, [viewport, lensAngleState]);
+
+  // Handle distortion, saturation and vignette
+  useEffect(() => {
+    if (!viewport)
```

Comment:
When switching between models seems like everything except Lens Angle is kept.

Reply:
Also in PROD. So not related to this PR. Can you create an Issue? We can come back to this one

Reply:
Issue opened.